provider/aws/aws.go

```diff
@@ -832,6 +850,23 @@ func (p *AWSProvider) AdjustEndpoints(endpoints []*endpoint.Endpoint) ([]*endpoi
 		} else {
 			ep.DeleteProviderSpecificProperty(providerSpecificEvaluateTargetHealth)
 		}
+
+		// Add provider specific properties to the endpoint if record has set identifier
+		if ep.SetIdentifier != "" {
+
+			// check if ep has any of the geoproximity properties
+			_, ok1 := ep.GetProviderSpecificProperty(providerSpecificGeoProximityLocationAWSRegion)
+			_, ok2 := ep.GetProviderSpecificProperty(providerSpecificGeoProximityLocationLocalZoneGroup)
+			_, ok3 := ep.GetProviderSpecificProperty(providerSpecificGeoProximityLocationCoordinates)
+
+			if ok1 || ok2 || ok3 {
+				// check if ep has bias property
+				if _, ok := ep.GetProviderSpecificProperty(providerSpecificGeoProximityLocationBias); !ok {
+					// if not, set it to 0
+					ep.SetProviderSpecificProperty(providerSpecificGeoProximityLocationBias, "0")
+				}
+			}
```

Comment:
If there is a more idiomatic way to express this, please let me know. I am not yet very experienced with Go.

Reply:
But basically what is happening here is, when a record with geoproximity routing is read from r53, it will always have a value for _Bias_. If the _Bias_ annotation is missing on the source, then the actual endpoint and the source endpoint don't match. That leads to unnecessary Upserts. 
IIUC, the AdjustEndpoints function is used to "normalize" endpoints and avoid such unnecessary changes.